pkg/apis/servicecatalog/validation/instance_test.go

```diff
@@ -141,3 +142,124 @@ func TestValidateInstance(t *testing.T) {
 		}
 	}
 }
+
+func TestValidateInstanceUpdate(t *testing.T) {
+	cases := []struct {
+		name  string
+		old   *servicecatalog.Instance
+		new   *servicecatalog.Instance
+		valid bool
+		err   string // Error string to match against if error expected
+	}{
+		{
+			name: "no update with async op in progress",
+			old: &servicecatalog.Instance{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-instance",
+					Namespace: "test-ns",
+				},
+				Spec: servicecatalog.InstanceSpec{
+					ServiceClassName: "test-serviceclass",
+					PlanName:         "Test-Plan",
+				},
+				Status: servicecatalog.InstanceStatus{
+					AsyncOpInProgress: true,
+				},
+			},
+			new: &servicecatalog.Instance{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-instance",
+					Namespace: "test-ns",
+				},
+				Spec: servicecatalog.InstanceSpec{
+					ServiceClassName: "test-serviceclass",
+					PlanName:         "Test-Plan",
+				},
+				Status: servicecatalog.InstanceStatus{
+					AsyncOpInProgress: true,
+				},
+			},
+			valid: false,
+			err:   "Another operation for this service instance is in progress",
+		},
+		{
```

Comment:
One thing I realized while reviewing this code (not a comment about your PR): if you look at `InstanceRestStrategy.PrepareForUpdate`, we don't allow _any_ update of the spec right now, so we will need to take that out once we begin implementing plan and parameter updates.